lib/galaxy/jobs/__init__.py

```diff
@@ -1115,6 +1115,10 @@ def can_split(self):
         # Should the job handler split this job up?
         return self.app.config.use_tasked_jobs and self.tool.parallelism
 
+    @property
+    def is_cwl_job(self):
+        return self.tool.tool_type == "cwl"
```

Comment:
```suggestion
        return self.tool and self.tool.tool_type == "cwl"
```
I suppose if you uninstall a tool while a job exists you may run into trouble.